reports/spec4-alloy.tex

```diff
@@ -1,7 +1,144 @@
+that satisfy simple properties. For example:
+
+\begin{lstlisting}[language=alloy,columns=fullflexible]
+  run someFinalizedCheckpoint { some c: Checkpoint |
+    isFinalized[c] and c.slot != 0
+  }
+  for 10 but 6 Block, 6 Checkpoint, 12 Vote, 5 int
+\end{lstlisting}
+
+For small search scopes, Alloy was quickly finding examples, often in a matter
+of seconds.
+
+Ultimately, we are interested in showing that~$\textit{AccountableSafety}$
+holds true. To this end, we want Alloy to show that the
+formula~$\textit{noAccountableSafety}$ does not have a model (unsatisfiable).
+Similar to Apalache, Alloy allows us to show safety for bounded scopes.
+
+\begin{lstlisting}[language=alloy,columns=fullflexible]
+  pred noAccountableSafety {
+    disagreement and (3.mul[#slashableNodes] < #Signature)
+  }
+  run noAccountableSafety for 10 but 6 Block, 6 Checkpoint,
+                                     4 Signature, 6 FfgVote, 15 Vote, 5 int
+\end{lstlisting}
+
+Our goal was to see, how far Alloy could go, when we were increasing the scope.
+Table~\ref{tab:alloy-mc} summarizes our experiments with Alloy when checking
+$\textit{noAccountableSafety}$ with Kissat. As we can see, we have managed to
+show accountable safety for the chain configurations of up to 6 blocks,
+including the genesis. However, the most interesting configuration of 7 blocks
+(on two chains) happens to be quite challenging for Alloy and Kissat.
+
+\begin{table}
+    \centering
+    \begin{tabular}{rrrrrrrr}
+        \tbh{\#}
+            & \tbh{blk}
+            & \tbh{chk}
+            & \tbh{sig}
+            & \tbh{ffg}
+            & \tbh{votes}
+            & \tbh{Time}
+            & \tbh{Memory}
+            \\ \toprule
+        1 & 3 & 5 & 4 & 5  & 12 & 4s & 35M
+            \\ \midrule
+        2 & 4 & 5 & 4 & 5 & 12 & 10s & 40M
+            \\ \midrule
+        3 & 5 & 5 & 4 & 5 & 12 & 15s & 45M
+            \\ \midrule
+        4 & 3 & 6 & 4 & 6 & 15 & 57s & 52M
+            \\ \midrule
+        5 & 4 & 6 & 4 & 6 & 15 & 167s & 55M
+            \\ \midrule
+        6 & 5 & 6 & 4 & 6 & 15 & 245s & 57M
+            \\ \midrule
+        7 & 6 & 6 & 4 & 6 & 15 & 360s & 82M
+            \\ \midrule
+        8 & 5 & 7 & 4 & 6 & 24 & 1h 27m & 156M
+            \\ \midrule
+        9 & --- & --- & --- & --- & --- & --- & ---
+            \\ \midrule
+        10 & 3 & 15 & 4 & 5 & 12 & 31s & 56M
+            \\ \midrule
+        11 & 4 & 20 & 4 & 5 & 12 & 152s & 94M
+            \\ \midrule
+        12 & 5 & 25 & 4 & 5 & 12 & 234s & 117M
+            \\ \midrule
+        13 & 7 & 15 & 4 & 10 & 40 & over 10 days & 300 MB
+            \\ \bottomrule
+    \end{tabular}
+    \caption{Model checking experiments with Alloy and Kissat:
+      \textbf{blk} is the number of blocks, \textbf{chk} is the number
+      of checkpoints, \textbf{sig} is the number of validators' signatures,
+      \textbf{ffg} is the number of FFG votes, \textbf{votes} is the number
```

Comment:
Maybe add a footnote that `ffg` bounds the number of FFG votes we would see in `votes`?